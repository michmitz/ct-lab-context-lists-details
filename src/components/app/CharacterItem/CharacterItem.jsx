import React from 'react';
import PropTypes from 'prop-types';
import { useColorMode } from '../../../hooks/colormode'

const CharacterItem = ({ name, imageUrl }) => {
  const { colorMode } = useColorMode();

  return <figure className={colorMode}>
    <img src={imageUrl} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>
}

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
}

export default CharacterItem