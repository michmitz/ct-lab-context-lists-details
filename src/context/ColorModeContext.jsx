import React from 'react';
import { useState } from 'react';

export const ColorModeContext = React.createContext();

export const ColorModeProvider = ({ children }) => {
  const [colorMode, setColorMode] = useState('light');

  const toggle = () => {
    if(colorMode === 'light') setColorMode('dark');
    if(colorMode === 'dark') setColorMode('light');
  };

  const bucket = {
    colorMode,
    toggle
  };

  return (
    <ColorModeContext.Provider value={bucket}>
      {children}
    </ColorModeContext.Provider>
  );
};