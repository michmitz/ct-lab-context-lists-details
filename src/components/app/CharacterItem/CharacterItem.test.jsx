import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';
import { ColorModeProvider } from '../../../context/ColorModeContext'

describe('CharacterItem component', () => {
  afterEach(() => cleanup());

  it('renders single character', () => {
    const { asFragment } = render(
    <ColorModeProvider>
    <CharacterItem
      name='Michelle' 
      imageUrl='url@url.com'
    />
    </ColorModeProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});