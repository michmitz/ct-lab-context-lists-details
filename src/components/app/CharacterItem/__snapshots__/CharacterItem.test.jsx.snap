// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders single character 1`] = `
<DocumentFragment>
  <figure
    class="light"
  >
    <img
      alt="Michelle"
      src="url@url.com"
    />
    <figcaption>
      Michelle
    </figcaption>
  </figure>
</DocumentFragment>
`;
